pkg/provision/providers/docker/node.go

```diff
@@ -84,24 +86,35 @@ func (p *provisioner) createNode(ctx context.Context, clusterReq provision.Clust
 			"talos.cluster.name": clusterReq.Name,
 			"talos.type":         nodeReq.Type.String(),
 		},
-		Volumes: map[string]struct{}{
-			"/var/lib/containerd": {},
-			"/var/lib/kubelet":    {},
-			"/etc/cni":            {},
-			"/run":                {},
-			"/system":             {},
-		},
 	}
 
 	// Create the host config.
 
+	mounts := []mount.Mount{}
+
+	for _, path := range []string{"/run", "/system", "/tmp"} {
+		mounts = append(mounts, mount.Mount{
+			Type:   mount.TypeTmpfs,
+			Target: path,
+		})
+	}
+
+	for _, path := range append(talosmount.Overlays, "/var", "/system/state") {
+		mounts = append(mounts, mount.Mount{
+			Type:   mount.TypeVolume,
+			Target: path,
+		})
+	}
```

Comment:
Mimic the `overlay` and other persistent mounts we create in non-container mode.